### PR TITLE
User ip cluster cron

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@yingjieMiao does this look right?

I had difficulty writing the receive function in the Plugin. The function signature is really strange.

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/shoebox/cron/UserIpAddressClusterCron.scala [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import com.keepit.common.plugin.{ SchedulerPlugin, SchedulingProperties }
+import com.keepit.common.service.IpAddress
+import com.keepit.common.time._
+import com.keepit.model.User
+import com.keepit.shoebox.cron.UserIpAddressClusterAction.{ DailySearchForClusters, WeeklySearchForClusters }
+import us.theatr.akka.quartz.QuartzActor
+
+trait UserIpAddressClusterCronPlugin extends SchedulerPlugin
+
+class UserIpAddressClusterCronPluginImpl @Inject() (
```
Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/shoebox/cron/UserIpAddressClusterCron.scala:38]: we normally have `@Singleton` annotation on plugins. c.f. `UriIntegrityPlugin` and `UriIntegrityPluginImpl`

File: kifi-backend/modules/shoebox/app/com/keepit/shoebox/cron/UserIpAddressClusterCron.scala [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  object DailySearchForClusters
+  object WeeklySearchForClusters
+}
+
+class UserIpAddressClusterActor @Inject() (
+    userIpAddressCommander: UserIpAddressCommander,
+    airbrake: AirbrakeNotifier) extends FortyTwoActor(airbrake) with Logging {
+  private val ipClusterSlackChannel = "#useripclusters"
+
+  def formatClusters(clusters: Seq[(IpAddress, Int, Id[User])]): String = {
```
Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/shoebox/cron/UserIpAddressClusterCron.scala:63]: could just be `private`?

**Comment by yingjieMiao** [comment]
looks good! merge when you feel it's ready.

File: kifi-backend/modules/shoebox/app/com/keepit/shoebox/cron/UserIpAddressClusterCron.scala (hunk 2 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/shoebox/cron/UserIpAddressClusterCron.scala:63]: Good call.

**Pull request merged**